aptos-move/block-executor/src/view.rs

```diff
@@ -816,59 +866,50 @@ impl<'a, T: Transaction> ResourceState<T> for SequentialState<'a, T> {
         target_kind: ReadKind,
         layout: UnknownOrLayout,
         patch_base_value: &dyn Fn(&T::Value, Option<&MoveTypeLayout>) -> PartialVMResult<T::Value>,
-    ) -> ReadResult {
+    ) -> PartialVMResult<ReadResult> {
         match self.unsync_map.fetch_data(key) {
             Some(mut value) => {
                 // If we have a known layout, upgrade RawFromStorage value to Exchanged.
                 if let UnknownOrLayout::Known(layout) = layout {
                     if let ValueWithLayout::RawFromStorage(v) = value {
-                        match patch_base_value(v.as_ref(), layout) {
-                            Ok(patched_value) => {
-                                let exchanged_value = ValueWithLayout::Exchanged(
-                                    Arc::new(patched_value.clone()),
-                                    layout.cloned().map(Arc::new),
-                                );
-                                self.unsync_map
-                                    .set_base_value(key.clone(), exchanged_value.clone());
+                        let patched_value = patch_base_value(v.as_ref(), layout)?;
```

Comment:
We no longer mark incorrect use if patching fails? I guess this should be ok

Reply:
fixed.